genesis/process/genesisBlockCreator.go

```diff
@@ -218,6 +227,53 @@ func (gbc *genesisBlockCreator) CreateGenesisBlocks() (map[uint32]data.HeaderHan
 	return genesisBlocks, nil
 }
 
+// in case of hardfork initial smart contracts deployment is not called as they are all imported from previous state
+func (gbc *genesisBlockCreator) computeDNSAddressesIfHardFork() error {
+	var dnsSC genesis.InitialSmartContractHandler
+	for _, sc := range gbc.arg.SmartContractParser.InitialSmartContracts() {
+		if sc.GetType() == genesis.DNSType {
+			dnsSC = sc
+			break
+		}
+	}
+
+	if dnsSC == nil || check.IfNil(dnsSC) {
+		return nil
+	}
+
+	builtInFuncs := builtInFunctions.NewBuiltInFunctionContainer()
+	argsHook := hooks.ArgBlockChainHook{
+		Accounts:         gbc.arg.Accounts,
+		PubkeyConv:       gbc.arg.PubkeyConv,
+		StorageService:   gbc.arg.Store,
+		BlockChain:       gbc.arg.Blkc,
+		ShardCoordinator: gbc.arg.ShardCoordinator,
+		Marshalizer:      gbc.arg.Marshalizer,
+		Uint64Converter:  gbc.arg.Uint64ByteSliceConverter,
+		BuiltInFunctions: builtInFuncs,
+	}
+	blockChainHook, err := hooks.NewBlockChainHookImpl(argsHook)
+	if err != nil {
+		return err
+	}
+
+	isForCurrentShard := func([]byte) bool {
```

Comment:
Perhaps add an inline comment to explain this, for future reference?

Reply:
added comment